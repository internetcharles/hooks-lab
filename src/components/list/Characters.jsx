import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character';

const Characters = ({ characters }) => {
  const characterElements = characters.map(character => (
    <li key={character.name}>
      <a key={character.name} href={`/${character.name}`}>
        <Character {...character} />
      </a>
    </li>
  ));

  return (
    <ul data-testid='list'>
      {characterElements}
    </ul>
  );
};

Characters.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    occupation: PropTypes.string.isRequired,
  }))
};

export default Characters;