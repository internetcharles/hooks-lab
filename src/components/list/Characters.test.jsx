import React from 'react';
import { render } from '@testing-library/react';
import Characters from './Characters';

describe('ListPage', () => {
  it('renders ListPage', () => {
    const { asFragment } = render(<Characters
      characters={[
        { 
          name: "John",
          image: "John.jpg",
          occupation: "Guy",
        }
      ]}  
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});