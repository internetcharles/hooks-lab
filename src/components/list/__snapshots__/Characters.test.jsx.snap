// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPage renders ListPage 1`] = `
<DocumentFragment>
  <ul
    data-testid="list"
  >
    <li>
      <a
        href="/John"
      >
        <figure>
          <h1>
            John
          </h1>
          <img
            alt="John"
            src="John.jpg"
          />
          <h3>
            Guy
          </h3>
        </figure>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
